Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/dialog/DialogFragmentTemplate.kt

```diff
@@ -0,0 +1,41 @@
+package de.rki.coronawarnapp.ui.dialog
+
+import android.app.Dialog
+import android.content.DialogInterface
+import android.os.Bundle
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.util.ContextExtensions.getColorCompat
+
+class DialogFragmentTemplate(
+    private val cancelable: Boolean = true,
+    private val isDeleteDialog: Boolean = false,
+    private val dismissAction: () -> Unit = { },
+    private val materialDialog: MaterialAlertDialogBuilder? = null,
+    private val config: MaterialAlertDialogBuilder.() -> Unit
+) : DialogFragment() {
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        isCancelable = cancelable
+        return materialDialog?.create() ?: MaterialAlertDialogBuilder(requireContext()).apply(config).create()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        if (isDeleteDialog) {
+            (dialog as AlertDialog).getButton(DialogInterface.BUTTON_POSITIVE)
+                ?.setTextColor(requireContext().getColorCompat(R.color.colorTextDeleteButtonDialog))
+        }
+    }
+
+    override fun onDismiss(dialog: DialogInterface) {
+        dismissAction()
+        super.onDismiss(dialog)
+    }
```

Comment:
I think  whole `onDismiss` is not needed and `dismissAction()` can be removed and `MaterialAlertDialogBuilder.setOnDismissListener` can be used as other functions

Reply:
I will try it again to make sure, but it didn't work for me. From reading up, it seems that when using dialogs inside a `DialogFragment`, this action is being handled by the `DialogFragment` itself 🤷 https://stackoverflow.com/a/18269965/13296736

Reply:
Tried it again, it dosen't work.

Reply:
I did try it and it worked 😄 , it might be because you were not applying the `config` when `dialog` is passed?